unit-tests-jest - Daniel Yount

TheDanYount opened the pull request.

TheDanYount committed:
Practiced unit testing

RobertGardner commented on unit-tests-jest/numbers.test.ts:
```diff
@@ -0,0 +1,81 @@
+});
+
+describe('divideBy', () => {
+  it('does not modify the original input array', () => {
```
Logically, you would test the happy path first, then the exceptions and side-effect behaviors like this. So I would put this one at or near the bottom of the list.

RobertGardner commented on unit-tests-jest/numbers.test.ts:
```diff
@@ -0,0 +1,81 @@
+import * as functions from './numbers';
```
Why not just import each individual function?

TheDanYount replied:
Personal preference for brevity, but I see how one might want to list the names of the functions one imported for reference.

RobertGardner commented on unit-tests-jest/numbers.test.ts:
```diff
@@ -0,0 +1,81 @@
+    const obj = {
+      firstName: 'Bob',
+      fakeNum: '5',
+      zeroEquivalent: null,
```
Nice.

RobertGardner commented on unit-tests-jest/numbers.test.ts:
```diff
@@ -0,0 +1,81 @@
+    const multiplier = 2;
+    const result = functions.multiplyBy(obj, multiplier);
+    expect(result).toEqual({});
+  });
```
You could also conceivably write a test that makes sure that `obj` is modified, since that's what the spec says.

RobertGardner reviewed (approved):
Minor comments. Good!

TheDanYount committed:
Re-ordered code to make it more readable

TheDanYount committed:
Added missing test

Added a test to make sure an object is being changed and returned, rather than a new object being substituted for it.
